andela/colossus-backend
# #167733810 A JWT Is Sent Back On Successful Registration

#### What does this PR do?
This PR implements the signup functionality, enabling users to register for the service

#### Description of Task to be completed?
- Setup the user route in the server file
- Create the user file in the routes folder
- Create a user file in the controller's folder and import it for use in the route folder
- Set up the user controller to implement the signup algorithm
- Document the feature using Swagger

#### How should this be manually tested?
- Ensure node.js and postman are installed
- Clone the repository to your system
- Navigate to the just cloned repo on your command line
- Type "npm install"
- Set environment variables DATABASE_DEV(database connection url) JWT_SECRET(any secret)
- Then type "npm run start-dev"
- Once the server starts running, use postman to send a POST request to the server endpoint
/api/v1/auth/signup. The request body should contain the fields: firstName, lastName, email, and password
- On successful registration a JWT is sent in the response body with other details as well

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- #167733810
